src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/ReclassifyCallbackService.java

```diff
@@ -0,0 +1,119 @@
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.envelopes.model.Classification.NEW_APPLICATION;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.envelopes.model.Classification.SUPPLEMENTARY_EVIDENCE_WITH_OCR;
+
+@Service
+public class ReclassifyCallbackService {
+
+    private static final Logger log = LoggerFactory.getLogger(ReclassifyCallbackService.class);
+
+    private static final String REQUIRED_INITIAL_STATE = "ScannedRecordReceived";
+    private static final String REQUIRED_INITIAL_CLASSIFICATION = NEW_APPLICATION.name();
+
+    public ProcessResult reclassifyExceptionRecord(CaseDetails exceptionRecordDetails, String userId) {
+        Long exceptionRecordId = exceptionRecordDetails.getId();
+        String jurisdiction = exceptionRecordDetails.getJurisdiction();
+
+        log.info(
+            "Reclassifying exception record. Exception record ID: {}. Jurisdiction: {}. User ID: {}",
+            exceptionRecordId,
+            jurisdiction,
+            userId
+        );
+
+        List<String> errors = validateCallbackRequest(exceptionRecordDetails);
+
+        if (errors.isEmpty()) {
+            log.info(
+                "Returning successful reclassification result for exception record. "
+                    + "Exception record ID: {}. Jurisdiction: {}. User ID: {}",
+                exceptionRecordId,
+                jurisdiction,
+                userId
+            );
+
+            return new ProcessResult(updateClassification(exceptionRecordDetails.getData()));
+        } else {
+            // Logging validation failure as error, because the caseworker should have no way of causing it.
+            // If it happens, the team needs to be alerted.
+            log.error(
+                "Validation failed for exception record reclassification. "
+                + "Exception record ID: {}. Jurisdiction: {}. User ID: {}. Errors: {}",
+                exceptionRecordId,
+                jurisdiction,
+                userId,
+                String.join(", ", errors)
+            );
+
+            return new ProcessResult(emptyList(), errors);
+        }
+    }
+
+    private Map<String, Object> updateClassification(Map<String, Object> exceptionRecordData) {
+        Map<String, Object> updatedFields = Maps.newHashMap(exceptionRecordData);
+        updatedFields.put(JOURNEY_CLASSIFICATION, SUPPLEMENTARY_EVIDENCE_WITH_OCR.name());
+        return updatedFields;
+    }
+
+    private List<String> validateCallbackRequest(CaseDetails exceptionRecordDetails) {
+        List<Optional<String>> validationErrors = asList(
+            validateState(exceptionRecordDetails),
+            validateClassification(exceptionRecordDetails)
+        );
+
+        return validationErrors
+            .stream()
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(toList());
```

Comment:
Spring has a nice wrapper called `Optionals`, it could transform this piece of code to something as following:

```java
return Optionals
    .toStream(
        alidateState(exceptionRecordDetails),
        alidateClassification(exceptionRecordDetails)
    )
    .collect(toList());
```

Reply:
Nice one, will do

Reply:
... although we would need to import a new library for this.

Reply:
ah. possibly it's badly implemented as i think it comes from data. tested on send letter project - had that window open. ignore then. unless orchestrator has smth handy from its' dependencies